### PR TITLE
Ensure consistency between iOS and Android SDK X docs

### PR DESCRIPTION
**Pull request by jaydeep-helpshift**

### CODE REVIEW

File: docs/sdkx_android/migration-guide.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 sidebar_position: 140
 title: Migration Guide
-description: "SDK X is a Hybrid SDK that helps you roll out innovation faster by enabling most of the updates over the air that flow to end users without any downtime or app updates."
```
Comment (sumeet-helpshift) [inline docs/sdkx_android/migration-guide.mdx:4]: I think this string was provided by product in migration guide.
We should check once before removing completely.

Comment (jaydeep-helpshift) [reply docs/sdkx_android/migration-guide.mdx:4]: This string is still present in the Intro section of this page. I thought it made the description too wordy, so removed it.

**Review by IamJoyson — APPROVED** [review]

**Review by sumeet-helpshift — APPROVED** [review]

File: docs/sdkx_android/delegates.mdx [hunk 2]
```diff
@@ -304,68 +283,69 @@ Reason type :
...
-        switch(eventName){
-          case HelpshiftEvent.RECEIVED_UNREAD_MESSAGE_COUNT:
-            int count = (int) data.get(HelpshiftEvent.DATA_MESSAGE_COUNT);
-            boolean fromCache = (boolean) data.get(HelpshiftEvent.DATA_MESSAGE_COUNT_FROM_CACHE);
-            Log.d(TAG, "Message Count : " + count + ", From Cache: " + fromCache);
-        }
-      }
-    // ...
-```
+If you want to retrieve the current notification count stored locally, you can call the same method with the parameter set to false, `requestUnreadMessageCount(false)`. In this case, SDK returns the count of unread messages available locally in this delegate method.
```
Comment (sumeet-helpshift) [inline docs/sdkx_android/delegates.mdx:349]: "requestUnreadMessageCount" prefix with "Helpshift. requestUnreadMessageCount(false)"

File: docs/sdkx_android/tracking.mdx [hunk 3]
```diff
@@ -122,75 +133,69 @@ If you want to set Custom Issue Fields at the time of Issue creation, follow the
...
+| ------------ | ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| "singleline" | string        | Single line string with character limit of 255                                                                                                                |
+| "multiline"  | string        | Multi line string with character limit of 100,000                                                                                                             |
+| "number"     | string        | String representation of number. For eg. "12345"                                                                                                              |
+| "dropdown"   | string        | Drop-down options should exist for the given Custom Issue Field on dashboard. Value should be one of the values specified on the dashbaord for that dropdown. |
+| "date"       | long long int | Epoch time in milliseconds. For eg. `1505927361535`                                                                                                           |
```
Comment (sumeet-helpshift) [inline docs/sdkx_android/tracking.mdx:144]: "long long int", should be just long right ?

File: docs/sdkx_android/tracking.mdx [hunk 4]
```diff
@@ -122,75 +133,69 @@ If you want to set Custom Issue Fields at the time of Issue creation, follow the
...
+| "checkbox"   | string        | String representation of boolean. For eg. "true" or "false". This corresponds to the checkbox type custom issue field on dashboard.                           |
```
Comment (sumeet-helpshift) [inline docs/sdkx_android/tracking.mdx:145]: we need boolean for checkbox right ?

File: docs/sdkx_android/tracking.mdx [hunk 5]
```diff
@@ -122,75 +133,69 @@ If you want to set Custom Issue Fields at the time of Issue creation, follow the
...
+HSDebugLog.d("TAG", "Any Debug log string");
 ```
 
-Debug logs are collected within the set debug logs limit. The limit is set under the SDK Configurations section for App Settings in the Helpshift’s Agent Dashboard. Debug logs are collected in a FIFO queue.
+Debug logs are collected within the set debug logs limit. The limit is set under the **SDK Configurations** section for App Settings in the Helpshift’s Agent Dashboard. Debug logs are collected in a FIFO queue.
+
+<Admonition type="info" title="Note">
+
+- Applicable to SDK X v10.1.0 & above
+- In the `leaveBreadcrumb` API, passing of formatted strings is NOT supported. It is recommended to pass data only in a simple String format. For example:
```
Comment (sumeet-helpshift) [inline docs/sdkx_android/tracking.mdx:197]: "leaveBreadcrumb", this should be HSDebug logs api

File: docs/sdkx_android/users.mdx [hunk 6]
```diff
@@ -102,33 +106,42 @@ You should call Helpshift SDK's Login API whenever the user successfully logs in
...
 
 #### Overview
 
 Once a user logs out from your app, you should call the Helpshift SDK logout API to ensure no one else can view this user's conversations.
-_Note:_ The logout API is expected to be used in conjunction with the login API.
 
-#### How to use
+<Admonition type="info" title="Note">
+
+- The logout API is expected to be used in conjunction with the login API.
```
Comment (sumeet-helpshift) [inline docs/sdkx_android/users.mdx:132]: We can also add a note here for having unique user id across all the users of the customer and that the user id should not be duplicated for 2 different users

File: docs/sdkx_ios/delegates.mdx [hunk 7]
```diff
@@ -33,215 +33,229 @@ All the public APIs in the SDK should be called after initializing the SDK via [
...
-
-This event gets fired when the Helpshift session starts
+### Widget Toggle Event {#widgetToggle}
 
-Event name: `HelpshiftEventNameSessionStarted` <br />
-Event data: `nil`
+This event is triggered when the user opens or exits the chat screen. This event is triggered with a boolean value of `"visible"` key. For your reference, see the below example:
 
-Example:
+- Event name: `HelpshiftEvent.WIDGET_TOGGLE`
```
Comment (IamJoyson) [inline docs/sdkx_ios/delegates.mdx:154]: This is android events in iOS it could be, the below example code looks good
1. HelpshiftEventNameWidgetToggle
2. HelpshiftEventDataVisible

File: docs/sdkx_ios/delegates.mdx [hunk 8]
```diff
@@ -467,146 +441,145 @@ func handleHelpshiftEvent(_ eventName: String, withData data: [AnyHashable : Any
...
 
-### Conversation Reopened Event {#conversationReopened}
+### Unread Message Count Event {#unreadMessageCount}
 
-When resolution question is enabled, the users are asked if they're satisfied with the resolution. If the user rejects it and sends a new message, then the conversation is reopened and the Conversation Reopened event is triggered.
+If you want a count of all new messages received in an existing conversation, you can call this API `[Helpshift requestUnreadMesssageCount:];`.
 
-Event name: `HelpshiftEventNameConversationReopened` <br />
-Event Data: `nil`
+The count of the number of messages will be conveyed to your app via this event. You can also use this event to keep your badge counts updated.
```
Comment (IamJoyson) [inline docs/sdkx_ios/delegates.mdx:549]: Can we change this to
The unread message count will be conveyed to your app via this event. You can also use this event to keep your badge counts updated.

OR 

The count of unread messages will be conveyed to your app via this event. You can also use this event to keep your badge counts updated.

File: docs/sdkx_ios/getting-started.mdx [hunk 9]
```diff
@@ -1,7 +1,7 @@
 ---
 sidebar_position: 10
 title: Getting Started
-description: "Getting started with integrating Helpshift in your app."
+description: Follow these simple steps to add Helpshift in-app support to your iOS app right away.
```
Comment (IamJoyson) [inline docs/sdkx_ios/getting-started.mdx:4]: In inorder the get this as a subtitle I think we should also update line 26. which indicates Getting started with integrating Helpshift in your app. but in Android it is.

Follow these simple steps to add Helpshift in-app support to your
Android app right away

File: docs/sdkx_android/getting-started.mdx [hunk 10]
```diff
@@ -122,8 +127,3 @@ Calling any API before the install call would throw an unchecked HelpshiftInitia
 Calling `install()` below android SDK version 21 will throw this checked exception. All the APIs will be non operable.
 
 </Admonition>
-
```
[the remark this answers is not in the document]
Comment (IamJoyson) [reply docs/sdkx_android/getting-started.mdx:125]: line 127 
android small case to Android

**Review by IamJoyson — CHANGES_REQUESTED** [review]

File: docs/sdkx_ios/notifications.mdx [hunk 11]
```diff
@@ -1,7 +1,7 @@
 ---
 sidebar_position: 50
 title: Notifications
-description: "Configure Push and In-app notifications."
+description: "Learn more about configuring push and in-app notifications."
```
[the remark this answers is not in the document]
Comment (IamJoyson) [reply docs/sdkx_ios/notifications.mdx:4]: line 519 sentence starts with a small `to retrieve the unread messages count stored locally and display the unread count similar to mentioned in the above examples.`

**Review by IamJoyson — CHANGES_REQUESTED** [review]
reviewed till notification page.

File: docs/sdkx_ios/outbound-support.mdx [hunk 12]
```diff
@@ -89,114 +86,58 @@ To pass the outbound support data to Helpshift, follow these steps -
...
-            [Helpshift handleNotificationWithUserInfoDictionary:notification.request.content.userInfo isAppLaunch:NO withController:self.window.rootViewController];
+- (void) userNotificationCenter:(UNUserNotificationCenter *)center
+    willPresentNotification:(UNNotification *)notification
+    withCompletionHandler:(void (^)(UNNotificationPresentationOptions))completionHandler {
+    if([[notification.request.content.userInfo objectForKey:@"origin"] isEqualToString:@"helpshift"]) {
+        // This is a notification from Helpshift push notification system
+        [Helpshift handleNotificationWithUserInfoDictionary:notification.request.content.userInfo isAppLaunch:NO withController:self.window.rootViewController];
+        completionHandler(UNNotificationPresentationOptionNone);
+    } else {
+        NSString* proactiveLink = [[notification.request.content.userInfo objectForKey:@"aps"] objectForKey:@"helpshift_proactive_link"];
```
Comment (IamJoyson) [inline docs/sdkx_ios/outbound-support.mdx:104]: Can we update this example, currently we are suggesting reading the data from "aps" object which is not the recomded. if users are using additional package something aps object will not be passed, because aps has info on notification sound, type etc.. 

Checked the apple website they suggest to put custom data outside of the aps object. 
https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/generating_a_remote_notification#2943359

File: docs/sdkx_ios/sdk-configuration.mdx [hunk 13]
```diff
@@ -49,6 +47,8 @@ Upon setting `enableLogging` to `YES`, Helpshift SDK logs will be generated in t
 | Min SDK       | v10.0.0                                           |
 | Supported by: | <code>installWithPlatformId:domain:config:</code> |
 
+</div>
```
Comment (IamJoyson) [inline docs/sdkx_ios/sdk-configuration.mdx:50]: The below example code for install API
on android <PLATFORM_ID>, <DOMAIN>
iOS objective-c YOUR_PLATFORM_ID, YOUR_DOMAIN_NAME
iOS swit it is YOUR_PLATFORM_ID, YOUR_DOMAIN

File: docs/sdkx_ios/sdk-configuration.mdx [hunk 14]
```diff
@@ -122,33 +114,77 @@ Helpshift.showFAQs(self, config: config)
 
 </Tabs>
 
-### Best Practices
+#### Best Practices
 
 - Provide tier-based support by setting enableContactUs to `ALWAYS` for paid users and `AFTER_VIEWING_FAQS` for unpaid ones.
 - Provide country based support by setting enableContactUs to `ALWAYS` for local users and `AFTER_VIEWING_FAQS` for foreign ones. Example Code:
 
 ```objc
```
Comment (IamJoyson) [inline docs/sdkx_ios/sdk-configuration.mdx:122]: we could also give swift code for the below example.

File: docs/sdkx_ios/sdk-configuration.mdx (hunk 13 above)
Comment (IamJoyson) [reply docs/sdkx_ios/sdk-configuration.mdx:50]: This is repeated in all code snippet in the page.

**Review by IamJoyson — CHANGES_REQUESTED** [review]

File: docs/sdkx_android/tracking.mdx (hunk 3 above)
Comment (jaydeep-helpshift) [reply docs/sdkx_android/tracking.mdx]: Right. Fixed.

File: docs/sdkx_android/tracking.mdx (hunk 4 above)
Comment (jaydeep-helpshift) [reply docs/sdkx_android/tracking.mdx:144]: Not according to code snippet above.

File: docs/sdkx_ios/migration-guide.mdx [hunk 15]
```diff
@@ -1,7 +1,7 @@
 ---
 sidebar_position: 140
 title: Migration Guide
-description: "SDK X is a Hybrid SDK that helps you roll out innovation faster by enabling most of the updates over the air that flow to end users without any downtime or app updates. This migration guide will walk you through the steps you need to take in order to migrate from Helpshift legacy SDK to SDK X."
+description: "This migration guide will walk you through the steps you need to take in order to migrate from Helpshift legacy SDK to SDK X."
```
[the remark this answers is not in the document]
Comment (IamJoyson) [reply docs/sdkx_ios/migration-guide.mdx:4]: iOS was not supporting this
